internal/connectors/engine/engine.go

```diff
@@ -76,6 +82,14 @@ func (e *engine) InstallConnector(ctx context.Context, provider string, rawConfi
 		Config:    rawConfig,
 	}
 
+	// Detached the context to avoid being in a weird state if request is
+	// cancelled in the middle of the operation.
+	ctx, _ = contextutil.Detached(ctx)
+	// Since we detached the context, we need to wait for the operation to finish
+	// even if the app is shutting down gracefully.
+	e.wg.Add(1)
+	defer e.wg.Done()
+
```

Comment:
Detatching the context should probably be after the entity is installed in the DB (only to ensure that the workflow is triggered). We still want the cancellation context to be passed to the DB driver. If there are multiple inserts that take place, we can use a rollback to revert the change.

Reply:
I'm not sure, imagine that the service is stopped between the connectors install and before we add in the waitgroup, we will be in a weird state, am I wrong ?